pytorch_lightning/core/step_result.py

```diff
@@ -306,111 +306,116 @@ def __copy__(self):
 
     @classmethod
     def gather(cls, outputs):
-        meta = outputs[0].get('meta')
+        meta = outputs[0].get("meta")
         result = cls()
         result = recursive_gather(outputs, result)
         recursive_stack(result)
 
         if meta:
-            result['meta'] = meta
+            result["meta"] = meta
         return result
 
     @classmethod
     def padded_gather(cls, outputs):
-        meta = outputs[0].get('meta')
+        meta = outputs[0].get("meta")
         result = cls()
         result = recursive_gather(outputs, result)
 
         # find the padding used for other values
         default_padding_idx = 0
         for name, value in result.items():
             if isinstance(value, list) and len(value) > 0 and isinstance(value[0], torch.Tensor):
-                if name not in {'checkpoint_on', 'early_stop_on', 'minimize'}:
-                    default_padding_idx = meta[name]['tbptt_pad_token']
+                if name not in {"checkpoint_on", "early_stop_on", "minimize"} and name in meta:
+                    default_padding_idx = meta[name]["tbptt_pad_token"]
                     break
 
         # pad across each key individually
         for name, value in result.items():
-            is_reserved = name in {'checkpoint_on', 'early_stop_on', 'minimize'}
-            if isinstance(value, list) and len(value) > 0 and isinstance(value[0], torch.Tensor):
-
+            is_reserved = name in {"checkpoint_on", "early_stop_on", "minimize"}
+            if isinstance(value, list) and len(value) > 0 and isinstance(value[0], torch.Tensor) and name in meta:
                 if is_reserved:
                     padding_key = default_padding_idx
                 else:
-                    padding_key = meta[name]['tbptt_pad_token']
+                    padding_key = meta[name]["tbptt_pad_token"]
                 padded = torch.nn.utils.rnn.pad_sequence(value, batch_first=True, padding_value=padding_key)
                 result[name] = padded
 
                 # also update the result
                 if meta and not is_reserved:
-                    meta[name]['value'] = padded
+                    meta[name]["value"] = padded
+
+        # collate tensors that are not in meta (e.g. user assigned attributes)
+        for name, value in result.items():
+            if name not in meta:
+                result[name] = collate_tensors(result[name])
+
```

Comment:
This is the biggest change, user assigned attributes must be collated separately as they are not in the `meta` dict.